src/applications/model/lognorm3-random-variable.h

```diff
@@ -0,0 +1,98 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#ifndef LOG_NORMAL3_RANDOM_VARIABLE_H
+#define LOG_NORMAL3_RANDOM_VARIABLE_H
+
+#include "ns3/random-variable-stream.h"
+
+namespace ns3 {
+
+/**
+ * \ingroup randomvariable
+ * \brief The three-parameters log-normal distribution Random Number Generator
+ * (RNG) that allows stream numbers to be set deterministically.
+ *
+ * The three-parameter log-normal distribution is simply the usual two-parameter
+ * log-normal distribution with a location shift.
+ * Let X be a random variable with a three-parameter log-normal distribution
+ * with parameters meanlog=mu, sdlog=sigma, and threshold=thresh
+ * Then the random variable Y=X−γ has a log-normal distribution with parameters
+ * meanlog=mu and sdlog=sigma
+ */
+class LogNormal3RandomVariable : public LogNormalRandomVariable
+{
+public:
+  /**
+   * \brief Register this type.
+   * \return The object TypeId.
+   */
+  static TypeId GetTypeId (void);
+
+  /**
+   * \brief Creates a log-normal distribution RNG with the default
+   * values for mu, sigma and thresh.
+   */
+  LogNormal3RandomVariable ();
+
+  /**
+   * \brief Returns the threshold value for the three-parameters log-normal distribution returned by this RNG stream.
+   * \return The threshold value for the three-parameters log-normal distribution returned by this RNG stream.
+   */
+  double GetThreshold (void) const;
+
+  /**
+   * \brief Returns a random double from a log-normal distribution with the specified mu and sigma.
+   * \param [in] thresh Sigma value for the three-parameters log-normal distribution.
+   * \return A floating point random value.
+   */
+  double GetValue (double thresh);
+
+  /**
+   * \brief Returns a random unsigned integer from a three-parameters log-normal distribution with the specified mu and sigma.
+   * \param [in] thresh Sigma value for the three-parameters log-normal distribution.
+   * \return A floating point random value.
+   */
+  uint32_t GetInteger (uint32_t thresh);
+
+  /**
+   * \brief Returns a random double from a three-parameters log-normal distribution with the current mu and sigma.
+   * \return A floating point random value.
+   */
+  virtual double GetValue (void);
+
+  /**
+   * \brief Returns a random unsigned integer from a three-parameters log-normal distribution with the current mu, sigma and thresh.
+   * \return A random unsigned integer value.
+   *
+   */
+  virtual uint32_t GetInteger (void);
+
+private:
+  /** The threshold value for the three-parameter log-normal distribution returned by this RNG stream. */
+  double m_thresh;
```

Comment:
Maybe let's call this variable `m_threshold` to be consistent with other code.

Reply:
agree